core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for learner answer info.
+ */
+
+require(
+  'pages/exploration-player-page/services/answer-classification.service.ts');
+require('pages/exploration-player-page/services/exploration-engine.service.ts');
+require('pages/exploration-player-page/services/player-transcript.service.ts');
+
+
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('LearnerAnswerInfoService', [
+  'AnswerClassificationService', 'ExplorationEngineService',
+  'PlayerTranscriptService', 'INTERACTION_IDS_WITHOUT_ANSWER_DETAILS',
+  'PROBABILITY_INDEXES',
+  function(
+      AnswerClassificationService, ExplorationEngineService,
+      PlayerTranscriptService, INTERACTION_IDS_WITHOUT_ANSWER_DETAILS,
+      PROBABILITY_INDEXES) {
+    var submittedAnswerInfoCount = 0;
+    var thresholdProbabilityIndex = null;
+    var randomProbabilityIndex = null;
+    var expId = null;
+    var exploration = null;
+    var state = null;
+    var stateName = null;
+    var interactionId = null;
+    var defaultOutcome = null;
+    var currentAnswer = null;
+    var currentInteractionRulesService = null;
+    var canAskLearnerForAnswerInfo = false;
+    var visitedStates = [];
+
+
+    var getRandomProbabilityIndex = function() {
+      var min = 0;
+      var max = 100;
+      return (Math.floor(Math.random() * (max - min + 1) ) + min) / 100;
+    };
+
+    return {
+      initLearnerForAnswerInfo: function(
+          answer, interactionRulesService) {
+        currentAnswer = answer;
+        currentInteractionRulesService = interactionRulesService;
+        expId = ExplorationEngineService.getExplorationId();
+        exploration = ExplorationEngineService.getExploration();
+        stateName = PlayerTranscriptService.getLastStateName();
+        state = exploration.getState(stateName);
```

Comment:
This similar method is being used in ExplorationEngineService

Reply:
I asked for these changes as I wanted to make this directive generic so that we can use it for questions as well.  Can we add state (or part of the state) as a parameter for the `initLearnerAnswerInfo` method?

Reply:
Okay @DubeySandeep I am trying to do that.

Reply:
Done